### DISCUSSION TITLE
Missing classes compiling to production

### DISCUSSION THREAD

**Discussion by msassa** [tailwindlabs/tailwindcss#9937] [Help] [answered]
Hi. I have a project in which I compile the assets for dev, everything works just fine. But when I do the same thing, but for production, there is one missing class, and I can't get why.

webpack.mix.js
```
const tailwindcss = require('tailwindcss');
require('laravel-mix-purgecss');

mix
    .js('resources/assets/js/app.js', 'public/js')
    .sass('resources/assets/sass/app.scss', 'public/css/app.css')
    .js('resources/assets/js/webcomponents/sliders.js', 'public/js/webcomponents/')
    .sass('resources/assets/sass/webcomponents/sliders.scss', 'public/css/webcomponents/sliders.css')
    .options({
        processCssUrls: false,
        postCss: [tailwindcss()],
    })
    .purgeCss({
        whitelistPatterns: [/^slick/],
    })
    .vue({ version: 2 })
    .sourceMaps();
```

postcss.config.js
```
module.exports = {
  plugins: {
    tailwindcss: { config: './tailwind.config.js' },
  },
}
```

tailwind.config.js
```
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    "./storage/framework/views/*.php",
    "./resources/**/*.blade.php",
    "./resources/**/*.js",
    "./resources/**/*.vue",
    "./resources/assets/js/webcomponents/components/carImages.vue",
    "./public/js/webcomponents/sliders.js",
  ],
  safelist: [
    'h-[225px]',
  ],
  theme: {
    extend: {},
  },
  plugins: [],
  prefix: 'ws-',
}
```

The problem is with the h-[225px] class. I use that class in the carImages.vue component, and when I do `npm run dev` the class is present in the sliders.css, but when I run `npm run prod` the class is not present.
I even try to add that class to the safelist, but the result is the same.

Can anyone helpme with this, please?
Thanks

**Comment by wongjn** [comment, marked as answer]
Seems like it could be `laravel-mix-purgecss` purging the class out. As far as I am aware, the Tailwind PostCSS plugin does not change its output depending on development or production, whereas on the otherhand, a cursory glance at the [`laravel-mix-purgecss` NPM package page](https://www.npmjs.com/package/laravel-mix-purgecss) states:

> By default, PurgeCSS only works when building assets for production.

This could account for the difference in development and production builds.

**Reply by msassa** [reply]
YES! was that .. thank you!!
